### PR TITLE
TS Composition API: Mention explicit PropType on prop runtime declaration

### PR DESCRIPTION
**Pull request by julian-eckhardt**
## Description of Problem

Right now, the guide does not mention a way to have explicit prop type declarations and default values at the same time.
The only (close) solution, Reactivity Transform, is experimental and opt-in.

## Proposed Solution

Mentioning the `as PropType <Foo>` syntax, as in this PR.

## Additional Information

I'm filing this PR not only to contribute to the docs and hopefully help others, but also to verify my own approach on the problem. I'd be keen on maintainers' input on this.

### CODE REVIEW

File: src/guide/typescript/composition-api.md [hunk 1]
```diff
@@ -22,6 +22,22 @@ props.bar // number | undefined
 
 This is called "runtime declaration", because the argument passed to `defineProps()` will be used as the runtime `props` option.
 
+Prop types can also be explicitly declared when using runtime declaration:
+
+```vue
+<script setup lang="ts">
+const props = defineProps({
+  foo: {
+    type: String as PropType<string>,
```
Comment (tony19) [inline src/guide/typescript/composition-api.md:31]: This is a redundant typing as the type `string` can already be inferred from the `String` constructor (same for `number` and `Number` below). I recommend showing an example where `PropType` is necessary. For example:

```typescript
type: Array as PropType<string[]>
```

**Comment by skirtles-code** [comment]
There is a related issue for this: <https://github.com/vuejs/docs/issues/1621>.

File: src/guide/typescript/composition-api.md (hunk 1 above)
Comment (skirtles-code) [reply src/guide/typescript/composition-api.md:31]: Maybe reuse the `PropType` example from the Options API page, ported to `<script setup>`?

Comment (julian-eckhardt) [reply src/guide/typescript/composition-api.md:31]: @tony19 I know, but for `Object` that wouldn't be the case. I just wanted to show how it works, as a 1:1 copy of the previous example.

**Review by tony19 — APPROVED** [review]

**Comment by skirtles-code** [comment]
I gave this some more thought and decided it might be better to take a slightly different approach to documenting this. There are more details on #1621, which has now been merged.

I'm going to close this PR, but if anyone has any further thoughts on this topic then feel free to share them on #1621 or open a new issue.

Thanks.

**Pull request closed without merging**